<launch>
  <arg name="data_path" default="$(env HOME)/autoware_data" description="packages data and artifacts directory path"/>

  <arg name="input/image" default="~/image_raw"/>
  <arg name="input/rois" default="~/rois"/>
  <arg name="output/traffic_signals" default="classified/traffic_signals"/>
  <arg name="param_path" default="$(find-pkg-share autoware_traffic_light_classifier)/config/pedestrian_traffic_light_classifier.param.yaml" description="classifier param path"/>
  <arg name="model_path" default="$(var data_path)/traffic_light_classifier/ped_traffic_light_classifier_mobilenetv2_batch_6.onnx" description="path to pedestrian model"/>
  <arg name="label_path" default="$(var data_path)/traffic_light_classifier/lamp_labels_ped.txt" description="path pedestrian label"/>
  <arg name="build_only" default="false" description="exit after trt engine is built"/>

  <node pkg="autoware_traffic_light_classifier" exec="traffic_light_classifier_node" name="traffic_light_classifier" output="screen">
    <remap from="~/input/image" to="$(var input/image)"/>
    <remap from="~/input/rois" to="$(var input/rois)"/>
    <remap from="~/output/traffic_signals" to="$(var output/traffic_signals)"/>
    <param from="$(var param_path)" allow_substs="true"/>
    <param name="model_path" value="$(var model_path)"/>
    <param name="label_path" value="$(var label_path)"/>
    <param name="build_only" value="$(var build_only)"/>
  </node>
</launch>
